modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/utils.ts

```diff
@@ -318,6 +341,57 @@ function getUncategorizedPredefinedPosition(
 	return {x, y};
 }
 
+export function getUnsupportedRelationshipErrorMessage(
+	nodes: Node<ObjectDefinitionNodeData>[],
+	sourceNode: Node<ObjectDefinitionNodeData>,
+	targetNode: Node<ObjectDefinitionNodeData>
+) {
+	if (sourceNode.data?.system && targetNode.data?.system) {
+		return {
+			errorMessage: Liferay.Language.get(
+				'system-objects-cannot-be-related'
+			),
+		};
+	}
+	else if (sourceNode.data?.linkedObjectDefinition) {
+		return {
```

Comment:
Hey @igor-franca, if you are returning all `if` statements you can completely remove all `else` statments from this code, it is unnecessary